mii/client.py

```diff
@@ -55,51 +74,189 @@ class MIIClient():
+                 port_map=None,
+                 deployment_tag=None):
+        super().__init__(deployments, host, port)
+        self.lb_stub = None
+        if port is not None:
+            channel = create_channel(host, port) if not self.channel else self.channel
+            self.lb_stub = modelresponse_pb2_grpc.DeploymentManagementStub(channel)
+        self.lb_config = lb_config
+        self.model_path = model_path
+        self.port_map = port_map if port_map is not None else {}
+        self.deployment_tag = deployment_tag
+
+    async def add_models_async(self, proto_request):
+        await getattr(self.lb_stub, "AddDeployment")(proto_request)
+
+    def add_models(self,
+                   task=None,
+                   model=None,
+                   deployment_name=None,
+                   enable_deepspeed=True,
+                   enable_zero=False,
+                   ds_config=None,
+                   mii_config={},
+                   deployments=[],
+                   deployment_type=DeploymentType.LOCAL,
+                   model_path=None,
+                   version=1):
+
+        _, deployments = validate_deployment(task=task,
+                                             model=model,
+                                             deployment_name=deployment_name,
+                                             enable_deepspeed=enable_deepspeed,
+                                             enable_zero=enable_zero,
+                                             ds_config=ds_config,
+                                             mii_config=mii_config,
+                                             deployment_tag=self.deployment_tag,
+                                             deployments=deployments,
+                                             deployment_type=deployment_type,
+                                             model_path=model_path,
+                                             version=version)
+
+        if not deployments:  #Empty deployment
+            return None
+
+        deps = {
+            getattr(deployment,
+                    mii.constants.DEPLOYMENT_NAME_KEY): deployment
+            for deployment in deployments
+        }
+        lb_config, self.port_map = allocate_processes(deps, self.port_map)
+        lb_enabled = True if len(self.deployments) else False
+        if self.lb_config is not None:
+            self.lb_config.replica_configs.extend(lb_config.replica_configs)
+        else:
+            self.lb_config = lb_config
+        for deployment in deployments:
+            self.deployments[getattr(deployment,
+                                     mii.constants.DEPLOYMENT_NAME_KEY)] = deployment
+        if self.model_path is None and deployment_type == DeploymentType.LOCAL:
+            self.model_path = mii.constants.MII_MODEL_PATH_DEFAULT
+        elif self.model_path is None and deployment_type == DeploymentType.AML:
+            model_path = "model"
```

Comment:
We don't necessarily support multi-model / adding deployments with AML, so we should error out if deployment type is AML.